src/meltano/cli/config.py

```diff
@@ -372,16 +373,30 @@ def set_(
     value: t.Any,
     store: str,
     interactive: bool,
+    from_file: t.TextIO,
 ):
     """Set the configurations' setting `<name>` to `<value>`."""
     if len(setting_name) == 1:
         setting_name = tuple(setting_name[0].split("."))
 
     interaction = InteractiveConfig(ctx=ctx, store=store, extras=False)
+
     if interactive:
         interaction.configure_all()
-    else:
-        interaction.set_value(setting_name=setting_name, value=value, store=store)
+        ctx.exit()
+
+    if from_file:
+        setting_name += (value,)
+        value = from_file.read()
```

Comment:
Wdyt about stripping any whitespace from the file contents?

```suggestion
        value = from_file.read().strip()
```

Reply:
Yes that could make sense